sdl_android/src/main/java/com/smartdevicelink/protocol/SdlProtocol.java

```diff
@@ -0,0 +1,1319 @@
+                                break;
+                            }
+                        }
+                    }
+                }
+            }
+        }else{
+            Log.d(TAG, transportRecord.toString() + " transport was NOT registered!");
+        }
+        //Notify any listeners for this secondary transport
+        List<ISecondaryTransportListener> listenerList = secondaryTransportListeners.remove(transportRecord.getType());
+        if(listenerList != null){
+            for(ISecondaryTransportListener listener : listenerList){
+                if(registered) {
+                    listener.onConnectionSuccess(transportRecord);
+                }else{
+                    listener.onConnectionFailure();
+                }
+            }
+        }
+
+        if(DebugTool.isDebugEnabled()){
+            printActiveTransports();
+        }
+    }
+
+    public void onTransportsConnectedUpdate(List<TransportRecord> transports){
+        Log.d(TAG, "Connected transport update");
+
+        //Temporary: this logic should all be changed to handle multiple transports of the same type
+        ArrayList<TransportType> connectedTransports = new ArrayList<>();
+        for(TransportRecord record: transports){
+            connectedTransports.add(record.getType());
+        }
+
+        if(connectedPrimaryTransport != null && !connectedTransports.contains(connectedPrimaryTransport.getType())){
+            //The primary transport being used is no longer part of the connected transports
+            //The transport manager callbacks should hadle the disconnect code
+            connectedPrimaryTransport = null;
+            return;
+        }
+
+        if(activeTransports.get(SessionType.RPC) == null){
+            //There is no currently active transport for the RPC service meaning no primary transport
+            TransportRecord preferredPrimaryTransport = getPreferredTransport(requestedPrimaryTransports,transports);
+            if(preferredPrimaryTransport != null) {
+                connectedPrimaryTransport = preferredPrimaryTransport;
+                startService(SessionType.RPC, (byte) 0x00, false);
+            }else{
+                onTransportNotAccepted("No transports match requested primary transport");
+            }
+            // return;
+        }else if(requiresHighBandwidth){
+            //If this app has a primary transport already but requires a high bandwidth transport
+            //to properly function, it is now time to register over that transport to be used
+            TransportRecord preferredSecondaryTransport = getPreferredTransport(requestedPrimaryTransports,transports);
+            if(preferredSecondaryTransport != null) {
+
+                if(iSdlProtocol != null) {
+                    Log.d(TAG, "Registering secondary transport!");
+                    registerSecondaryTransport(iSdlProtocol.getSessionId(), preferredSecondaryTransport);
```

Comment:
Not certain the reason right now, but it appears that this "Register Secondary Transport" message is being sent over the primary transport, when it should be sent over the secondary transport.